core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java

```diff
@@ -168,11 +168,14 @@ private UnderFileSystem getOrAddWithRecorder(AlluxioURI ufsUri,
       mCloser.register(fs);
       try {
         connectUfs(fs);
-      } catch (IOException e) {
+        tryUseFileSystem(fs, ufsUri.getPath());
+      } catch (Exception e) {
+        mUnderFileSystemMap.remove(key);
```

Comment:
prefer to put ufs into mUnderFileSystemMap only after we verify that ufs is working.

Reply:
done.